#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.4558835380705993 0.7094961538752219 0.5373875764776291 4.427537167247274
  position -8.265751880494488 18.98487035828992 1.0580288206935589
  follow "Hexabot"
  followType "None"
}
Background {
  skyColor [
    0.15 0.45 1
  ]
}
Robot {
  translation 0 0.0650742 0
  rotation -0.005415912021686196 0.9998263732215681 -0.017829506655512008 -2.9867553071795863
  children [
    Display {
      name "Path_Display"
    }
    Display {
      name "Lidar_Display"
    }
    HokuyoUtm30lx {
      translation 0 0.1 0
      rotation 0 1 0 -1.5708
    }
    Compass {
      name "Hexabot_Compass"
    }
    Gyro {
      name "Hexabot_Gyro"
    }
    GPS {
      name "Hexabot_GPS"
    }
    Camera {
      translation 0.1 -1.12008e-13 -1.49482e-07
      rotation 8.684229999951365e-07 -0.9999999999943995 3.2321499999818985e-06 1.5707853071795865
      children [
        DEF CAM_SHAPE Shape {
          appearance MattePaint {
            baseColor 0.7 0.1 0.1
          }
          geometry Cylinder {
            height 0.05
            radius 0.0125
          }
        }
      ]
      name "Hexabot_Camera"
      boundingObject USE CAM_SHAPE
      physics Physics {
      }
      lens Lens {
      }
      focus Focus {
      }
      recognition Recognition {
      }
    }
    DEF BODY_TRAN Transform {
      children [
        DEF BODY Shape {
          appearance MattePaint {
            baseColor 0.1 0.1 0.7
          }
          geometry Box {
            size 0.2 0.05 0.125
          }
        }
      ]
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        position -0.0022739080850000537
      }
      jointParameters2 JointParameters {
        position -0.22139096718981555
        axis 0 1 0
      }
      device [
        DEF SHOULDER_2_x RotationalMotor {
          name "Hexabot_Leg0_Motor1"
        }
      ]
      device2 [
        DEF SHOULDER_0_y RotationalMotor {
          name "Hexabot_Leg0_Motor2"
        }
      ]
      endPoint DEF SHOULDER_0 Solid {
        translation 0.07933303430706239 -0.011819710422135976 0.04807618155668188
        rotation 0.9893792621180315 0.01865511209716711 -0.14415499465305223 2.0730544217226203
        children [
          Transform {
            children [
              DEF FEMUR_0 Solid {
                translation 0 0.025981 -0.0450019
                rotation 1 0 0 0.523599
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position -0.23439476994565617
                      axis 0 0 1
                    }
                    device [
                      DEF ELBOW_0 RotationalMotor {
                        name "Hexabot_Leg0_Motor3"
                      }
                    ]
                    endPoint Solid {
                      translation -0.00013580694686209053 -0.0007227815181317747 -0.05856390241636175
                      rotation -0.6982496091365232 -0.7011123663735179 -0.14453004206329764 2.818487674021208
                      children [
                        Transform {
                          children [
                            Solid {
                              translation 0.0824507 0 0.0210637
                              rotation 0 -1 0 1.8325953071795864
                              children [
                                Shape {
                                  appearance MattePaint {
                                    baseColor 0.1 0.1 0.6
                                  }
                                  geometry DEF TIBIA_SHAPE Box {
                                    size 0.025 0.025 0.15
                                  }
                                }
                              ]
                              boundingObject USE TIBIA_SHAPE
                              physics Physics {
                              }
                            }
                            DEF ELBOW_SHAPE Shape {
                              appearance MattePaint {
                                baseColor 0 0 0
                              }
                              geometry Cylinder {
                                height 0.025
                                radius 0.0125
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE ELBOW_SHAPE
                      physics Physics {
                      }
                    }
                  }
                  DEF FEMUR_SHAPE Shape {
                    appearance MattePaint {
                      baseColor 0.2 0.2 1
                    }
                    geometry Box {
                      size 0.025 0.025 0.1
                    }
                  }
                ]
                boundingObject USE FEMUR_SHAPE
                physics Physics {
                }
              }
              Shape {
                appearance MattePaint {
                  baseColor 0 0 0
                }
                geometry DEF SHOULDER_SHAPE Sphere {
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "solid(5)"
        boundingObject USE SHOULDER_SHAPE
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        position 0.40227344074420135
      }
      jointParameters2 JointParameters {
        position -0.1786090556826884
        axis 0 1 0
      }
      device [
        DEF SHOULDER_2_x RotationalMotor {
          name "Hexabot_Leg1_Motor1"
        }
      ]
      device2 [
        DEF SHOULDER_0_y RotationalMotor {
          name "Hexabot_Leg1_Motor2"
        }
      ]
      endPoint DEF SHOULDER_0 Solid {
        translation 0.0002648227463395332 -0.014073133930614278 0.05816049349091708
        rotation 0.9952254293949537 0.0394993837794893 -0.08925325409569473 1.6857032508215701
        children [
          Transform {
            children [
              DEF FEMUR_0 Solid {
                translation 0 0.025981 -0.0450019
                rotation 1 0 0 0.523599
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position -0.3656051753975401
                      axis 0 0 1
                    }
                    device [
                      DEF ELBOW_0 RotationalMotor {
                        name "Hexabot_Leg1_Motor3"
                      }
                    ]
                    endPoint Solid {
                      translation -0.00021972105758464328 -0.0005873330376399193 -0.0586179508060048
                      rotation -0.7415673076921311 -0.6528377967375679 -0.1545339422686227 2.8379207097850694
                      children [
                        Transform {
                          children [
                            Solid {
                              translation 0.0824507 0 0.0210637
                              rotation 0 -1 0 1.8325953071795864
                              children [
                                Shape {
                                  appearance MattePaint {
                                    baseColor 0.1 0.1 0.6
                                  }
                                  geometry DEF TIBIA_SHAPE Box {
                                    size 0.025 0.025 0.15
                                  }
                                }
                              ]
                              boundingObject USE TIBIA_SHAPE
                              physics Physics {
                              }
                            }
                            DEF ELBOW_SHAPE Shape {
                              appearance MattePaint {
                                baseColor 0 0 0
                              }
                              geometry Cylinder {
                                height 0.025
                                radius 0.0125
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE ELBOW_SHAPE
                      physics Physics {
                      }
                    }
                  }
                  DEF FEMUR_SHAPE Shape {
                    appearance MattePaint {
                      baseColor 0.2 0.2 1
                    }
                    geometry Box {
                      size 0.025 0.025 0.1
                    }
                  }
                ]
                boundingObject USE FEMUR_SHAPE
                physics Physics {
                }
              }
              Shape {
                appearance MattePaint {
                  baseColor 0 0 0
                }
                geometry DEF SHOULDER_SHAPE Sphere {
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "solid(4)"
        boundingObject USE SHOULDER_SHAPE
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        position 1.1977252172191966
      }
      jointParameters2 JointParameters {
        position -0.22139143118087923
        axis 0 1 0
      }
      device [
        DEF SHOULDER_2_x RotationalMotor {
          name "Hexabot_Leg2_Motor1"
        }
      ]
      device2 [
        DEF SHOULDER_0_y RotationalMotor {
          name "Hexabot_Leg2_Motor2"
        }
      ]
      endPoint DEF SHOULDER_0 Solid {
        translation -0.08807183680369116 0.0003624775797416366 0.05029815885915852
        rotation 0.9947491342002581 -0.04864681186799204 -0.09004247721446312 2.2806829236536266
        children [
          Transform {
            children [
              DEF FEMUR_0 Solid {
                translation 0 0.025981 -0.0450019
                rotation 1 0 0 0.523599
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position -0.23439459063395232
                      axis 0 0 1
                    }
                    device [
                      DEF ELBOW_0 RotationalMotor {
                        name "Hexabot_Leg2_Motor3"
                      }
                    ]
                    endPoint Solid {
                      translation -0.00013722730150394637 -0.0006867865599434188 -0.05857588456507868
                      rotation -0.7035355976794022 -0.7087859294974512 -0.051576825651841844 3.0030948911187765
                      children [
                        Transform {
                          children [
                            Solid {
                              translation 0.0824507 0 0.0210637
                              rotation 0 -1 0 1.8325953071795864
                              children [
                                Shape {
                                  appearance MattePaint {
                                    baseColor 0.1 0.1 0.6
                                  }
                                  geometry DEF TIBIA_SHAPE Box {
                                    size 0.025 0.025 0.15
                                  }
                                }
                              ]
                              boundingObject USE TIBIA_SHAPE
                              physics Physics {
                              }
                            }
                            DEF ELBOW_SHAPE Shape {
                              appearance MattePaint {
                                baseColor 0 0 0
                              }
                              geometry Cylinder {
                                height 0.025
                                radius 0.0125
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE ELBOW_SHAPE
                      physics Physics {
                      }
                    }
                  }
                  DEF FEMUR_SHAPE Shape {
                    appearance MattePaint {
                      baseColor 0.2 0.2 1
                    }
                    geometry Box {
                      size 0.025 0.025 0.1
                    }
                  }
                ]
                boundingObject USE FEMUR_SHAPE
                physics Physics {
                }
              }
              Shape {
                appearance MattePaint {
                  baseColor 0 0 0
                }
                geometry DEF SHOULDER_SHAPE Sphere {
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "solid(3)"
        boundingObject USE SHOULDER_SHAPE
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        position -0.0022734351127279007
      }
      jointParameters2 JointParameters {
        position -0.2213909551329728
        axis 0 1 0
      }
      device [
        DEF SHOULDER_3_x RotationalMotor {
          name "Hexabot_Leg3_Motor1"
        }
      ]
      device2 [
        DEF SHOULDER_3_y RotationalMotor {
          name "Hexabot_Leg3_Motor2"
        }
      ]
      endPoint DEF SHOULDER_0 Solid {
        translation -0.08797362065009404 0.009944665602610439 -0.04103956471107509
        rotation -0.05618871659532283 0.9984190764542716 0.0014750930715817835 0.040472431101634884
        children [
          Transform {
            translation 0 0 -0.01
            children [
              DEF FEMUR_0 Solid {
                translation 0 0.0259808 -0.045
                rotation 1 0 0 0.523599
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position -0.23439482380192034
                      axis 0 0 1
                    }
                    device [
                      DEF ELBOW_3 RotationalMotor {
                        name "Hexabot_Leg3_Motor3"
                      }
                    ]
                    endPoint Solid {
                      translation -9.11512040675435e-06 -2.345422292386418e-05 -0.060020559987986694
                      rotation 3.9218853593379725e-05 5.435751167595619e-05 -0.9999999977535711 1.5433932167172113
                      children [
                        Transform {
                          children [
                            Solid {
                              translation 0.0824507 0 0.0210637
                              rotation 0 -1 0 1.8325953071795864
                              children [
                                Shape {
                                  appearance MattePaint {
                                    baseColor 0.1 0.1 0.6
                                  }
                                  geometry DEF TIBIA_SHAPE Box {
                                    size 0.025 0.025 0.15
                                  }
                                }
                              ]
                              boundingObject USE TIBIA_SHAPE
                              physics Physics {
                              }
                            }
                            DEF ELBOW_SHAPE Shape {
                              appearance MattePaint {
                                baseColor 0 0 0
                              }
                              geometry Cylinder {
                                height 0.025
                                radius 0.0125
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE ELBOW_SHAPE
                      physics Physics {
                      }
                    }
                  }
                  DEF FEMUR_SHAPE Shape {
                    appearance MattePaint {
                      baseColor 0.2 0.2 1
                    }
                    geometry Box {
                      size 0.025 0.025 0.1
                    }
                  }
                ]
                boundingObject USE FEMUR_SHAPE
                physics Physics {
                }
              }
              Shape {
                appearance MattePaint {
                  baseColor 0 0 0
                }
                geometry DEF SHOULDER_SHAPE Sphere {
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "solid(2)"
        boundingObject USE SHOULDER_SHAPE
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        position -0.09772427116499857
      }
      jointParameters2 JointParameters {
        position -0.17860906505240814
        axis 0 1 0
      }
      device [
        DEF SHOULDER_4_x RotationalMotor {
          name "Hexabot_Leg4_Motor1"
        }
      ]
      device2 [
        DEF SHOULDER_4_y RotationalMotor {
          name "Hexabot_Leg4_Motor2"
        }
      ]
      endPoint DEF SHOULDER_0 Solid {
        translation -0.010794512078465995 -0.00514004717564975 -0.05252492440260663
        rotation -0.47886135433888477 -0.8768427647800314 0.04287853977970074 0.20353335260857783
        children [
          Transform {
            translation 0 0 -0.01
            children [
              DEF FEMUR_0 Solid {
                translation 0 0.0259808 -0.045
                rotation 1 0 0 0.523599
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position 0.03439486529858127
                      axis 0 0 1
                    }
                    device [
                      DEF ELBOW_4 RotationalMotor {
                        name "Hexabot_Leg4_Motor3"
                      }
                    ]
                    endPoint Solid {
                      translation -8.530216196340773e-13 2.4791061366845604e-11 -0.0599999999999991
                      rotation 2.0671465643934447e-10 2.1394949867021575e-10 -1 1.5364004418808386
                      children [
                        Transform {
                          children [
                            Solid {
                              translation 0.0824507 0 0.0210637
                              rotation 0 -1 0 1.8325953071795864
                              children [
                                Shape {
                                  appearance MattePaint {
                                    baseColor 0.1 0.1 0.6
                                  }
                                  geometry DEF TIBIA_SHAPE Box {
                                    size 0.025 0.025 0.15
                                  }
                                }
                              ]
                              boundingObject USE TIBIA_SHAPE
                              physics Physics {
                              }
                            }
                            DEF ELBOW_SHAPE Shape {
                              appearance MattePaint {
                                baseColor 0 0 0
                              }
                              geometry Cylinder {
                                height 0.025
                                radius 0.0125
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE ELBOW_SHAPE
                      physics Physics {
                      }
                    }
                  }
                  DEF FEMUR_SHAPE Shape {
                    appearance MattePaint {
                      baseColor 0.2 0.2 1
                    }
                    geometry Box {
                      size 0.025 0.025 0.1
                    }
                  }
                ]
                boundingObject USE FEMUR_SHAPE
                physics Physics {
                }
              }
              Shape {
                appearance MattePaint {
                  baseColor 0 0 0
                }
                geometry DEF SHOULDER_SHAPE Sphere {
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "solid(1)"
        boundingObject USE SHOULDER_SHAPE
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        position 0.09772657351358532
      }
      jointParameters2 JointParameters {
        position -0.2213909849972539
        axis 0 1 0
      }
      device [
        DEF SHOULDER_5_x RotationalMotor {
          name "Hexabot_Leg5_Motor1"
        }
      ]
      device2 [
        DEF SHOULDER_5_y RotationalMotor {
          name "Hexabot_Leg5_Motor2"
        }
      ]
      endPoint DEF SHOULDER_0 Solid {
        translation 0.09122263576939404 0.004009822961255238 -0.04077188661700275
        rotation 0.40231009507828386 -0.9144107410540918 -0.04471670876762858 0.24192024385286204
        children [
          Transform {
            translation 0 0 -0.01
            children [
              DEF FEMUR_0 Solid {
                translation 0 0.0259808 -0.045
                rotation 1 0 0 0.523599
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position 0.16560516979867598
                      axis 0 0 1
                    }
                    device [
                      DEF ELBOW_5 RotationalMotor {
                        name "Hexabot_Leg5_Motor3"
                      }
                    ]
                    endPoint Solid {
                      translation 2.6181140124201026e-23 -1.5664584341642611e-22 -0.060000000000000095
                      rotation -1.2411177001124669e-20 -4.975922038976328e-21 -1 1.4051901373808513
                      children [
                        Transform {
                          children [
                            Solid {
                              translation 0.0824507 0 0.0210637
                              rotation 0 -1 0 1.8325953071795864
                              children [
                                Shape {
                                  appearance MattePaint {
                                    baseColor 0.1 0.1 0.6
                                  }
                                  geometry DEF TIBIA_SHAPE Box {
                                    size 0.025 0.025 0.15
                                  }
                                }
                              ]
                              boundingObject USE TIBIA_SHAPE
                              physics Physics {
                              }
                            }
                            DEF ELBOW_SHAPE Shape {
                              appearance MattePaint {
                                baseColor 0 0 0
                              }
                              geometry Cylinder {
                                height 0.025
                                radius 0.0125
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE ELBOW_SHAPE
                      physics Physics {
                      }
                      recognitionColors [
                        0 0 0
                      ]
                    }
                  }
                  DEF FEMUR_SHAPE Shape {
                    appearance MattePaint {
                      baseColor 0.2 0.2 1
                    }
                    geometry Box {
                      size 0.025 0.025 0.1
                    }
                  }
                ]
                boundingObject USE FEMUR_SHAPE
                physics Physics {
                }
              }
              Shape {
                appearance MattePaint {
                  baseColor 0 0 0
                }
                geometry DEF SHOULDER_SHAPE Sphere {
                  radius 0.02
                }
              }
            ]
          }
        ]
        boundingObject USE SHOULDER_SHAPE
        physics Physics {
        }
      }
    }
  ]
  name "Hexabot"
  boundingObject USE BODY_TRAN
  physics Physics {
  }
  controller "my_controller_c"
}
RectangleArena {
  translation 0 -0.05 0
  floorSize 10 10
  wallHeight 0.5
}
OilBarrel {
  translation -2.48 0.42 1.74
}
OilBarrel {
  translation -2.52 0.44 -1.79
  name "oil barrel(2)"
}
Wall {
  translation 0 -0.14 -4.69
  size 5 1 0.2
}
